struct.c

```diff
@@ -1295,18 +1316,27 @@ rb_struct_values_at(int argc, VALUE *argv, VALUE s)
 
 /*
  *  call-seq:
- *     struct.select {|obj| block }  -> array
- *     struct.select                 -> enumerator
- *     struct.filter {|obj| block }  -> array
- *     struct.filter                 -> enumerator
+ *    select {|value| ... } -> array
+ *    select -> enumerator
  *
- *  Yields each member value from the struct to the block and returns an Array
- *  containing the member values from the +struct+ for which the given block
- *  returns a true value (equivalent to Enumerable#select).
+ *  With a block given:
```

Comment:
This explanation is too long winded and goes into too much implementation detail. Maybe instead: `With a block given, returns an array of values from +self+ for which the block returns a truthy value`

Reply:
Done.